src/components/dialog/dialog.tsx

```diff
@@ -0,0 +1,814 @@
+import {
+  Component,
+  Element,
+  Event,
+  EventEmitter,
+  Host,
+  Watch,
+  Prop,
+  State,
+  h
+} from "@stencil/core";
+import { ChDialogResizeElement } from "./types";
+import { forceCSSMinMax, isRTL } from "../../common/utils";
+import { SyncWithRAF } from "../../common/sync-with-frames";
+
+// Custom vars
+const DIALOG_BLOCK_START = "--ch-dialog-block-start";
+const DIALOG_INLINE_START = "--ch-dialog-inline-start";
+const DIALOG_DRAGGED_X = "--ch-dialog-dragged-x";
+const DIALOG_DRAGGED_Y = "--ch-dialog-dragged-y";
+
+const DIALOG_RTL = "--ch-dialog-rtl";
+const DIALOG_RTL_VALUE = "-1";
+const DRAGGING_CLASS = "gx-dialog-dragging";
+const RESIZING_CLASS = "ch-dialog-resizing";
+
+const DIALOG_BLOCK_SIZE = "--ch-dialog-block-size";
+const DIALOG_INLINE_SIZE = "--ch-dialog-inline-size";
+
+const DIALOG_BORDER_BLOCK_START_SIZE = "--ch-dialog-border-block-start-width";
+const DIALOG_BORDER_BLOCK_END_SIZE = "--ch-dialog-border-block-end-width";
+const DIALOG_BORDER_INLINE_START_SIZE = "--ch-dialog-border-inline-start-width";
+const DIALOG_BORDER_INLINE_END_SIZE = "--ch-dialog-border-inline-end-width";
+
+const DIALOG_RTL_CLASS = "ch-dialog-rtl";
+
+const addCursorInDocument = (cursor: string) =>
+  (document.body.style.cursor = cursor);
+
+const resizingCursorDictionary: {
+  [key in ChDialogResizeElement]: (rtl: boolean) => void;
+} = {
+  "block-start": () => addCursorInDocument("ns-resize"),
+
+  "block-end": () => addCursorInDocument("ns-resize"),
+
+  "inline-start": () => addCursorInDocument("ew-resize"),
+
+  "inline-end": () => addCursorInDocument("ew-resize"),
+
+  "block-start-inline-start": rtl =>
+    addCursorInDocument(rtl ? "nesw-resize" : "nwse-resize"),
+
+  "block-start-inline-end": rtl =>
+    addCursorInDocument(rtl ? "nwse-resize" : "nesw-resize"),
+
+  "block-end-inline-start": rtl =>
+    addCursorInDocument(rtl ? "nwse-resize" : "nesw-resize"),
+
+  "block-end-inline-end": rtl =>
+    addCursorInDocument(rtl ? "nesw-resize" : "nwse-resize")
+};
+
+// Utils
+const fromPxToNumber = (pxValue: string) =>
+  Number(pxValue.replace("px", "").trim());
+
+const setProperty = (element: HTMLElement, property: string, value: number) =>
+  element.style.setProperty(property, `${value}px`);
+
+/**
+ * The `ch-dialog` component represents a modal or non-modal dialog box or other
+ * interactive component.
+ */
```

Comment:
Part names can be documented here. See for reference:
https://github.com/genexuslabs/chameleon-controls-library/blob/main/src/components/checkbox/checkbox.tsx#L43